File: buffer/gpu/src/jvmMain/java/com/wsr/gpu/elementwise/math/JMath.java
package com.wsr.gpu.elementwise.math;

public class JMath {
    public native void exp(long x, long result, long runtime);
    public native void ln(long x, float e, long result, long runtime);
    public native void sigmoid(long x, long result, long runtime);
    public native void pow(long x, int n, long result, long runtime);
    public native void sqrt(long x, float e, long result, long runtime);
}

File: buffer/gpu/src/jvmAndroidMain/kotlin/com/wsr/gpu/elementwise/math/JMath.kt
package com.wsr.gpu.elementwise.math

class JMath {
    external fun exp(x: Long, result: Long, runtime: Long)
    external fun ln(x: Long, e: Float, result: Long, runtime: Long)
    external fun sigmoid(x: Long, result: Long, runtime: Long)
    external fun pow(x: Long, n: Int, result: Long, runtime: Long)
    external fun sqrt(x: Long, e: Float, result: Long, runtime: Long)
}
